### PR TITLE
nixos/gnome: make it possible to remove core packages

### PR DESCRIPTION
**Pull request by rhoriguchi**
###### Motivation for this change

Make it possible to remove all packages which are default installed for gnome, for example `pkgs.gnome-tour` or `pkgs.gnome-user-docs`.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
Sorry, lost this in the noise. I resolved the conflict and moved around some stuff that should not be removable.

File: nixos/modules/services/x11/desktop-managers/gnome.nix [hunk 1]
```diff
@@ -400,6 +403,18 @@ in
...
+        [
+          pkgs.gnome.gnome-shell
+        ]
+        ++ (
+          utils.removePackagesByName
+            [
+              pkgs.gnome.gnome-shell-extensions
+            ]
+            config.environment.gnome.excludePackages
+        );
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/desktop-managers/gnome.nix:416]:
```suggestion
      services.xserver.desktopManager.gnome.sessionPath = [
        pkgs.gnome.gnome-shell
      ] ++ (utils.removePackagesByName [
        pkgs.gnome.gnome-shell-extensions
      ] config.environment.gnome.excludePackages);
```

File: nixos/modules/services/x11/desktop-managers/gnome.nix [hunk 2]
```diff
@@ -452,26 +467,32 @@ in
...
+              pkgs.gnome-tour # GNOME Shell detects the .desktop file on first log-in.
+              pkgs.gnome-user-docs
+              pkgs.orca
+              pkgs.glib # for gsettings program
+              pkgs.gnome-menus
+              pkgs.gtk3.out # for gtk-launch program
+              pkgs.xdg-user-dirs # Update user dirs as described in http://freedesktop.org/wiki/Software/xdg-user-dirs/
+            ]
+            config.environment.gnome.excludePackages
+        );
```
Comment (SuperSandro2000) [inline nixos/modules/services/x11/desktop-managers/gnome.nix:495]:
```suggestion
      environment.systemPackages = with pkgs.gnome; [
        gnome-shell
      ] ++ (utils.removePackagesByName [
        adwaita-icon-theme
        nixos-background-info
        gnome-backgrounds
        gnome-bluetooth
        gnome-color-manager
        gnome-control-center
        gnome-shell-extensions
        gnome-themes-extra
        pkgs.gnome-tour # GNOME Shell detects the .desktop file on first log-in.
        pkgs.gnome-user-docs
        pkgs.orca
        pkgs.glib # for gsettings program
        pkgs.gnome-menus
        pkgs.gtk3.out # for gtk-launch program
        pkgs.xdg-user-dirs # Update user dirs as described in http://freedesktop.org/wiki/Software/xdg-user-dirs/
      ] config.environment.gnome.excludePackages);
```

**Comment by jtojnar** [comment]
Rebased and switched the formatting to use `let` expression as recommended by K900.

**Pull request merged**